### DISCUSSION TITLE
[@types/react-native] Button is missing some props

### DISCUSSION THREAD

**Discussion by codyduong** [DefinitelyTyped/DefinitelyTyped#56288] [Issues with a @types package] [answered]
Here is the typing for the ButtonProps as it is currently typed
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/96d20a6a47593b83b0331a0a3f163a39aba523aa/types/react-native/index.d.ts#L7546-L7558

It looks like we are missing some types from https://reactnative.dev/docs/0.65/button,
is there a particular reason these props are missing (it seems these props are platform specific):

The following are missing:

- hasTVPreferredFocus
- nextFocusDown
- nextFocusForward
- nextFocusLeft
- nextFocusRight
- nextFocusUp

Theres also props which are not documented here but included in the react-native src code:
https://github.com/facebook/react-native/blob/main/Libraries/Components/Button.js#L262-L281
- accessibilityActions
- accessibilityHint
- onAccessibilityAction

Interestingly enough, we already type accessibilityState (despite it not actually being documented)

Docs for the accessibility typing can be found here: https://reactnative.dev/docs/0.65/accessibility#accessibility-actions

I think I have a pretty good understanding of this, so I can write a PR if this is the better intended typing.

[a comment by an automated account is not shown]

**Comment by ca057** [comment]
Hey :) I think the reason they’re missing is that now one got around yet to add the new typings and update them - so if you could make a PR this would be awesome!

**Comment by codyduong** [comment, marked as answer]
Oops, PR linked for posterity https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56374
